src/Microsoft.PowerShell.Commands.Utility/commands/utility/Measure-Object.cs

```diff
@@ -793,15 +830,41 @@ private MeasureInfo CreateGenericMeasureInfo(Statistics stat, bool shouldUseGene
         {
             double? sum = null;
             double? average = null;
+            double? stdDeviation = null;
             object max = null;
             object min = null;
 
             if (!_nonNumericError)
             {
                 if (_measureSum)
                     sum = stat.sum;
+
                 if (_measureAverage && stat.count > 0)
                     average = stat.sum / stat.count;
+
+                if(_measureStdDeviation && !_measureAverage) {
+                    ErrorRecord errorRecord = new ErrorRecord(
+                        PSTraceSource.NewArgumentException("Average"),
+                        "AverageSwitchNotSet",
+                        ErrorCategory.InvalidArgument,
+                        null);
+
+                    errorRecord.ErrorDetails = new ErrorDetails(this, "MeasureObjectStrings", "AverageSwitchNotSet", "Average");
+                    WriteError(errorRecord);
+                }
+
+                if (_measureStdDeviation && _measureAverage && stat.count > 0)
+                {
+                    var sumOfDerivation = 0.0;
+
+                    foreach (double n in stat.stdDeviationNumbers)
+                    {
+                        var m = n - (double)average;
+                        sumOfDerivation += m * m;
+                    }
+
+                    stdDeviation = Math.Round(Math.Sqrt(sumOfDerivation / (stat.stdDeviationNumbers.Count - 1)), 4);
```

Comment:
Please clarify why we round? I believe it must be a double.

Reply:
I rounded for no particular reason. I can remove it.